web/modules/custom/bnf/bnf_client/bnf_client.module

```diff
@@ -104,3 +124,36 @@ function bnf_client_form_node_form_submit(array $form, FormStateInterface $form_
 
   }
 }
+
+/**
+ * Implements hook_entity_base_field_info().
+ *
+ * Creating our custom programmatic "keep updated with BNF?" field.
+ *
+ * @return \Drupal\Core\Field\FieldDefinitionInterface[]
+ *   The field definitions.
+ */
+function bnf_client_entity_base_field_info(EntityTypeInterface $entity_type): array {
+  $fields = [];
+
+  $field_name = 'bnf_keep_updated';
+
+  // Create new fields for node bundle.
+  if ($entity_type->id() === 'node') {
+    $fields[$field_name] = BaseFieldDefinition::create('boolean')
```

Comment:
This is not what I had in mind.

`keep_updated` is not a state distinct from imported/exported, it is intrinsically bound to "imported".

We can just add `case LocallyClaimed = 3;` to `BnfStateEnum`, and the "backend" is basically implemented as the updating code only queues nodes where the state is `Imported`.

The form_alter then just add the checkbox when the state is either `LocallyClaimed` or `Imported`, and toggle between the two.

Reply:
What is the pros of doing it that way? Because, a con I see is that it makes it more confusing and complicated when we want to allow the editors to filter imported nodes in various views.

Reply:
Why would it make it more complicated? The pro is that we have guaranteed consistent data in one place. The backend doesn't have to check two fields in order to determine if it should sync. There's no risk of inconsistent data (what does "keep in sync" = true mean if status is none/exported?).

Reply:
Alright, you've got it your way - I've updated the code :)